### PR TITLE
worked on tags, tags single file, date modified and some scss stuff

### PR DESCRIPTION
**Pull request by SeniorMars**
- **fixed tiny bug with clipboard**
- **removed unneeded css**
- **worked on tags, tags single file, date modified and some scss stuff**
- **added feature to hide toc with click/hover**
- Also, ran linter on some files and edited markdown.md

Apologies since there are supposed to be separate PRs but I working on this at the same time.

Solves: #53

### CODE REVIEW

**Comment by SeniorMars** [comment]
I'll squash in a bit lol.

**Comment by SeniorMars** [comment]
this is awful, sorry. Please when you are about to github's ""Squash and Merge"". BUt this is good work.

File: sass/parts/_header.scss [hunk 1]
```diff
@@ -1,77 +1,112 @@
...
+    /* This changes the meta class to use flexbox, which ensures inline display */
+    align-items: center;
+    /* Aligns items vertically in the middle */
+    flex-wrap: wrap;
+    /* Allows items to wrap as needed */
+}
+
+.meta>time {
+    color: var(--primary-color);
 }
```
Comment (not-matthias) [inline sass/parts/_header.scss:103]: I'm not sure if I like this: 
image

Would you mind changing/removing this?

File: templates/macros/macros.html [hunk 2]
```diff
@@ -104,9 +103,25 @@ <h1 class="title">
                         Posted on <time>{{ page.date | date(format="%Y-%m-%d") }}</time>
                     {% endif %}
 
+                    {% if page.updated %}
+                        :: :odified on <time>{{ page.updated | date(format="%Y-%m-%d") }}</time>
```
Comment (not-matthias) [inline templates/macros/macros.html:107]: Small typo here, should be 'M'odified :)

**Review by not-matthias — CHANGES_REQUESTED** [review]

**Comment by not-matthias** [comment]
Thanks for all the new features, I really like them. Just some minor changes and then we can merge it :)

**Comment by SeniorMars** [comment]
Okay, so:

1. Modified the .meta tags to be something like this (with the pink removed!)
CleanShot 2024-06-18 at 10 13 40@2x

to enable something like this we need 

`repo_url = "https://github.com/SeniorMars/seniormars.com/blob/main/content/"` in config.toml and then

in your article:

```
[extra]
repo_view = true
read_time = true
```

this comes from `https://github.com/SeniorMars/redux/blob/0b6dcb48f74a24b4eaac9052db96d4ef6a984774/templates/macros/macros.html#L124` and `https://github.com/SeniorMars/redux/blob/0b6dcb48f74a24b4eaac9052db96d4ef6a984774/templates/macros/macros.html#L110`. I made sure to error handle.


 
 then I renamed `clipboard` to `fancy_code` as I realized that the code label also needed some js to move the label when the code is too long.
 
CleanShot 2024-06-18 at 10 17 58@2x

Furthermore, I added `h_anchor_link = true` which generates anchor tags for h1, h2, etc. I made sure to disable everything by default. 

Finally, I fixed some CSS and some unexpected behavior.

**Comment by not-matthias** [comment]
Looks good! Thanks :)

**Pull request merged**